app/common/reset.py

```diff
@@ -0,0 +1,39 @@
+import time
+from datetime import timedelta
+from secrets import token_urlsafe
+from typing import Any
+
+from cryptography.fernet import Fernet, InvalidToken, MultiFernet
+
+
+class TokenCryptographyProducer:
+    def __init__(self, fernet_key_list: list[str]) -> None:
+        self.encryptor = MultiFernet(
+            (Fernet(key.encode("utf-8")) for key in fernet_key_list)
+        )
+        self.token_randomness: int = 40
+        self.token_length: int = 15
+        self.token_ttl: timedelta = timedelta(hours=1)
+
+    def generate_token(self) -> str:
+        return token_urlsafe(self.token_randomness)[: self.token_length]
+
+    def encrypt_token(self, token: str) -> bytes:
+        token_bytes = token.encode()
+        encrypted_token = self.encryptor.encrypt_at_time(
+            msg=token_bytes, current_time=int(time.time())
+        )
```

Comment:
Так дока советует:
> The motivation behind this method is for the client code to be able to test token expiration. Since this method can be used in an insecure manner one should make sure the correct time (int(time.time())) is passed as current_time outside testing.

Reply:
Это не совет использовать этот метод, это пояснение, как он работает. Причин его использовать в цитате не написано

Reply:
Судя по их имплементации смысла особого нет, если не использовать mock-и на время (чего мы не делаем):

![image](https://github.com/xi-effect/xi.auth/assets/62603165/bc49bb79-684f-42f1-9639-742a90c0f541)